### PR TITLE
Luse/tozpatron 227/users backend integration

### PR DESCRIPTION
**Pull request by lsielewicz**
- Users backend integration

### CODE REVIEW

File: Toz.Dotnet/Toz.Dotnet/Controllers/UsersController.cs [hunk 1]
```diff
@@ -28,19 +28,21 @@ public class UsersController : Controller
...
         public async Task<IActionResult> Index(CancellationToken cancellationToken)
         {
-            List<User> users = await _usersManagementService.GetAllUsers();
+            List<User> users = await _usersManagementService.GetAllUsers(cancellationToken);
             return View(users);
         }
 
         public IActionResult Add()
         {
+            var usr = new User();
```
Comment (kkaniewski) [inline Toz.Dotnet/Toz.Dotnet/Controllers/UsersController.cs]: Why?

File: Toz.Dotnet/Toz.Dotnet/Controllers/UsersController.cs [hunk 2]
```diff
@@ -28,19 +28,21 @@ public class UsersController : Controller
...
+
             return PartialView(new User());
         }
 
         [HttpPost]
         [ValidateAntiForgeryToken]
         public async Task<IActionResult> Add(
-            [Bind("FirstName, LastName, PhoneNumber, Email, Purpose")] 
+            [Bind("Id, FirstName, LastName, PhoneNumber, Email, Password, Roles")]
```
Comment (kkaniewski) [inline Toz.Dotnet/Toz.Dotnet/Controllers/UsersController.cs]: On adding we don't have to pass ID I believe.

File: Toz.Dotnet/Toz.Dotnet/Views/Users/Add.cshtml [hunk 3]
```diff
@@ -46,19 +46,32 @@
...
+                        <option value="3">@Localizer["Administrator"]</option>
                     </select>
                     <span class="glyphicon form-control-feedback"></span>
-                    @Html.ValidationMessageFor(m => m.Purpose)
+                    @Html.ValidationMessageFor(m => m.Roles)
                 </div>
             </div>
+            
+            <div class="form-group row" style="text-align: center">
+                <span class="unhandled-error-span">@ViewData["UnhandledError"]</span>
```
Comment (kkaniewski) [inline Toz.Dotnet/Toz.Dotnet/Views/Users/Add.cshtml]: Doesn't show "unauthorized" error when I'm trying to assign somebody as administrator (Edit view works just fine)

File: Toz.Dotnet/Toz.Dotnet/Views/Users/Add.cshtml [hunk 4]
```diff
@@ -46,19 +46,32 @@
...
+            </div>
 
             <div class="form-group row">
                 <label class="col-sm-2 control-label">@Localizer["Purpose"]</label>
                 <div class="col-sm-10 selectContainer">
-                    <select asp-for="Purpose" class="form-control" id="Purpose">
-                        <option value="0">@Localizer["Volunteer"]</option>
-                        <option value="1">@Localizer["TemporaryHome"]</option>
-                        <option value="2">@Localizer["Administrator"]</option>
+                    <select asp-for="Roles" class="form-control" id="Roles" multiple="multiple">
```
Comment (kkaniewski) [inline Toz.Dotnet/Toz.Dotnet/Views/Users/Add.cshtml]: I'd prefer to use some kind of checkboxes list here if we have to do something like this (still no idea why BE allows multiple roles though)

File: Toz.Dotnet/Toz.Dotnet/Views/Users/Edit.cshtml [hunk 5]
```diff
@@ -53,24 +53,38 @@
...
+            </div>
 
             <div class="form-group row">
                 <label class="col-sm-2 control-label">@Localizer["Purpose"]</label>
                 <div class="col-sm-10 selectContainer">
-                    <select asp-for="Purpose" class="form-control" id="Purpose">
-                        <option value="0">@Localizer["Volunteer"]</option>
-                        <option value="1">@Localizer["TemporaryHome"]</option>
-                        <option value="2">@Localizer["Administrator"]</option>
+                    <select asp-for="Roles" class="form-control" id="Roles" multiple="multiple" overflow-y: auto>
```
Comment (kkaniewski) [inline Toz.Dotnet/Toz.Dotnet/Views/Users/Edit.cshtml]: "overflow-y: auto". I believe it's place is a CSS file :) (anyway auto makes no bigger sense)

File: Toz.Dotnet/Toz.Dotnet/Controllers/UsersController.cs (hunk 1 above)
Comment (lsielewicz) [inline Toz.Dotnet/Toz.Dotnet/Controllers/UsersController.cs]: ops, forgot to delete that thing (it was used for some check purpose I can't remember atm)

File: Toz.Dotnet/Toz.Dotnet/Controllers/UsersController.cs (hunk 2 above)
Comment (lsielewicz) [inline Toz.Dotnet/Toz.Dotnet/Controllers/UsersController.cs]: yop copy-pasta from editing, cuz I was too lazy to write "password" 🥇

File: Toz.Dotnet/Toz.Dotnet/Views/Users/Edit.cshtml (hunk 5 above)
Comment (lsielewicz) [inline Toz.Dotnet/Toz.Dotnet/Views/Users/Edit.cshtml]: fixed in 00fea18

File: Toz.Dotnet/Toz.Dotnet/Views/Users/Add.cshtml (hunk 3 above)
Comment (lsielewicz) [inline Toz.Dotnet/Toz.Dotnet/Views/Users/Add.cshtml]: fixed in 00fea18

File: Toz.Dotnet/Toz.Dotnet/Controllers/UsersController.cs (hunk 2 above)
Comment (lsielewicz) [inline Toz.Dotnet/Toz.Dotnet/Controllers/UsersController.cs]: fixed in 00fea18

File: Toz.Dotnet/Toz.Dotnet/Controllers/UsersController.cs (hunk 1 above)
Comment (lsielewicz) [inline Toz.Dotnet/Toz.Dotnet/Controllers/UsersController.cs]: fixed in 00fea18

**Comment by lsielewicz** [comment]
It's because we want to add some user without the all obligatory data (like password and email). We've decided to replace the current behaviour of that proccess to the simpler one. In future there won't be possible to create a new user while adding the reservation, it will be just a selecting the existing user from the list to assign him to the specified reservation.

That will be changed and fixed while impelementing the backend integration with scheduler.

**Pull request closed without merging**